File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()) {
            return false;
        }
        
        // solution 1
        // sort(s.begin(), s.end());
        // sort(t.begin(), t.end());
        // for(int i = 0; i < s.length(); i++)  {
        //     if(s[i] != t[i]) {
        //         return false;
        //     }
        // }
        // return true;
        
        // solution 2
        int freq[26] = {0};
        for(int i = 0; i < s.length(); i++) {
            freq[s[i] - 'a']++;
            freq[t[i] - 'a']--;
        }
        for(int i = 0; i < 26; i++) {
            if(freq[i] != 0) {
                return false;
            }
        }
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        charMap = [0 for _ in range(26)]
        for c in s:
            charMap[ord(c) - ord('a')] += 1
        for c in t:
            charMap[ord(c) - ord('a')] -= 1
        for c in charMap:
            if c:
                return False
        return True